### DISCUSSION TITLE
Can I change the Category Label dynamicly?

### DISCUSSION THREAD

**Discussion by rulai-yinqiang** [harborsiem/WindowsRibbon#29] [General]
I want to insert a table like Office, which means when the mouse hover a dropdown item, will also change the Category Label dynamicly. I tried many ways, the Category Label can not be refreshed.
![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/d7a934c3-cb88-4181-950f-cb38beb60692)

**Comment by harborsiem** [comment]
Which ways did you try? How did you detect the mouse hover?
I don't think that your request is possible.
The Ribbon Framework (WindowsRibbon) is different to the ribbon in Office.

**Comment by rulai-yinqiang** [comment]
In PreviewEvent, I can change the icon of the selected area. But I can not change the Category Label.
If I change the Category Label, it does not refresh.

I tried:
```
        Dim cat As Object
        iTable.Categories.GetItem(0, cat)
        cat.Label = "3 x 6 Table"
```
Also tried:
```
        Dim cat As Object
        cat = iTable.GalleryCategories(0)
        cat.Label = "3 x 6 Table"
```
But neither work, yhe Category Label does not refresh.

**Comment by harborsiem** [comment]
Add this after

```
cat.Label = "3 x 6 Table"
Ribbon.Framework.InvalidateUICommand(iTable.CommandID, RibbonLib.Interop.Invalidations.Property, PropertyKeyRef.From(RibbonProperties.Categories))

```
to refresh the UI.

I think it will be a good idea to add two new functions for all gallery controls (InvalidateCategories(), InvalidateItemsSource()) in the next version of the ribbon.

**Comment by rulai-yinqiang** [comment]
It can update the Category label with .InvalidateUICommand

But the whole controls in the category disappeared when update the Category label with .InvalidateUICommand.

**Comment by harborsiem** [comment]
I think it is not possible what you want to do with the Ribbon Framework. In the office ribbon it seems that there is an additional DropDown control. You can see this when you hover over the items. Here you see many items as selected. That is not possible in Ribbon Framework.

Maybe the disappeared area is a feature or a bug in the Microsoft implementation.
